src/SharedFixerConfig.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer;
+
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ */
+final class SharedFixerConfig
+{
+    private $indent;
+    private $lineEnding;
+
+    public function __construct($indent = '    ', $lineEnding = "\n")
+    {
+        $this->indent = $indent;
```

Comment:
valid cases:

indent: any single-line non-empty whitespaces of same type, so any positive number of tabs or any positive number of spaces, but not mixed

lineEnding: "\n" or "\r\n"